pkg/controller/deployment/sync.go

```diff
@@ -365,17 +365,6 @@ func (dc *DeploymentController) scale(deployment *extensions.Deployment, newRS *
 		return err
 	}
 
-	// If the new replica set is saturated, old replica sets should be fully scaled down.
-	// This case handles replica set adoption during a saturated new replica set.
-	if deploymentutil.IsSaturated(deployment, newRS) {
-		for _, old := range controller.FilterActiveReplicaSets(oldRSs) {
-			if _, _, err := dc.scaleReplicaSetAndRecordEvent(old, 0, deployment); err != nil {
-				return err
-			}
-		}
-		return nil
-	}
```

Comment:
I am not sure `NewRSNewReplicas` is the right place as that helper is called when we want to scale up or create a new RS (so the proportional scaling code is uncovered). Also scaleUpCount is always > 0 because maxTotalPods - currentPodCount > 0 and d.spec.replicas > newrs.spec.replicas.

I just realized that the real fix for the proportional scaling code is to avoid syncing replica sets with synced replicas annotations which we also need for cases when one of the RS updates fails [here](https://github.com/kubernetes/kubernetes/blob/94a5074bd6223d4e18f27eecba7892ae673f6c13/pkg/controller/deployment/sync.go#L446).